controllers/articles.controller.js

```diff
@@ -20,4 +32,72 @@ exports.getArticleById = (req, res, next) => {
             console.error(err)
             res.status(500).send({ msg: 'Internal Server Error' });
         });
-};
\ No newline at end of file
+};
+
+
+exports.getCommentsByArticle = (req, res, next) => {
+    const { article_id } = req.params
+    const regex = /^\d+$/;
+
+    if (!regex.test(article_id)) {
+        return res.status(400).send({ msg: "Invalid ID Format"})
+    }
+
+    checkIfArticleEXist(article_id)
+    .then((articleExist) => {
+        if (!articleExist) {
+            return Promise.reject({ status: 404, msg: 'Article Not Found'})
+        }
+        return fetchCommentsByArticle(article_id)
+    })
+        .then((comments) => {
+            res.status(200).send({ comments })
+        })
+        .catch((err) => {
+            next(err);
+        });
+};
+
+exports.addComment = (req, res, next) => {
+    const { article_id } = req.params
+    const { username, body } = req.body
+
+    if (isNaN(article_id)) {
+        return next({ status: 400, msg: 'Invalid ID Format'})
+    }
+    if (!username || !body) {
+        return next({ status: 400, msg:'Bad Request' })
+    }
+
+    insertComment(article_id, username, body)
+        .then((newComment) => {
+            res.status(201).send({ comment: newComment})
+        })
+        .catch((err) => {
+            if (err.code === '23503') {
+                return next({ status: 404, msg: 'Article or user not found'})
+            }
+            next(err);
+        })
+}
+
+exports.updateArticle = (req, res, next) => {
+    const { article_id } = req.params
+    const { inc_votes } = req.body
+
```

Comment:
These can be handled as PSQL errors not custom errors, read the notes on PSQL errors and investigate the err thrown in the controller catch using console.log